community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/RelationshipIdWrapperImpl.java

```diff
@@ -20,8 +20,10 @@
 package org.neo4j.cypher.internal.codegen;
 
 import org.neo4j.cypher.internal.compiler.v3_4.spi.RelationshipIdWrapper;
+import org.neo4j.values.AnyValueWriter;
+import org.neo4j.values.virtual.VirtualRelationshipValue;
 
-public final class RelationshipIdWrapperImpl implements RelationshipIdWrapper
+public final class RelationshipIdWrapperImpl extends VirtualRelationshipValue implements RelationshipIdWrapper
```

Comment:
Can't we just use NodeReference directly and get rid of the [Node,Relationship]Wrapper altogether?

Reply:
Yes definitely. I didn't get to that yet though.